framework/src/utils/MooseUtils.C

```diff
@@ -247,6 +254,15 @@ hasExtension(const std::string & filename, std::string ext, bool strip_exodus_ex
     return false;
 }
 
+void
+rmAll(const std::string & path)
+{
+  auto cmd = std::string("rm -rf ") + path;
```

Comment:
This is a bad idea... 

I would suggest several changes here:
 1. Validate the input. What happens if the input is blank, or worse "/", or "../", etc?
 2. Why do you need the force flag? We expect this to run in a directory that we create with our permissions. It should run without the force flag just fine. If it fails, fine, quite and say that you couldn't delete something.

IMHO, I think this is all just a bad design. I've fixed numerous bugs over the years related to reading/writing of checkpoint files with complex run arguments (MPI + threading) and we've found real issues related to the checkpoint system this way by figuring out threads were reading/writing the wrong files. The explicit delete lists of really helped! Note the deleting of the warnings about deleting individual files. We used to hit those until all of the bugs were resolved! This PR is removing all of that effort and going back to blindly and forcibly removing everything without even so much as validating the input. We'll not only delete files created by MOOSE but everything else too whether temporary or not, or with the right pattern or not. Yes, I agree that users shouldn't ever touch the contents of those files but that doesn't mean we should resort to this kind of removal strategy.

Reply:
The new checkpoint "files" are actually directories.  So the equivalent to "delete the individual file" became "delete the directory".  In the general case, we *can't* know what is in the directory - because that depends on what splits may have been generated for this mesh file potentially across separate runs of the application(s).

Reply:
If it is blank, it doesn't do anything.  Modern ``rm`` already has ``rm /`` protection built in.  And what about cases where someone wants to (on purpose) do ``rm ../foo/bar``?

Reply:
You are missing the point. Yes, I realize that you can't delete the root, but if somebody makes an error with the path, it wouldn't be difficult to zap an entire directory somewhere, possibly as high as the user's home directory. Consider the case where the guy was running (or attempting) to run MOOSE from his home directory. If the CWD was involved in anyway way and that path gets mangled what if we remove a good chunk of the user's home directory?

Reply:
Part of the removed warnings was because what were previously different file names based on parallel vs not are now in exactly the same self-contained checkpoint file structure regardless of whether running parallel or not - that accounts for one of the if/else branches.  One of the other warnings removed was because previously the header and main cpr files were together in the main directory and each needed a separate delete.  Now that they are contained in their own subdir, we don't need to delete them separately.  The removal of the ``-f`` from the rmAll implementation should replace the safety from the 3rd warning removal.

Reply:
> The new checkpoint "files" are actually directories. So the equivalent to "delete the individual file" became "delete the directory".

That's simply not correct. You've added a directory to the path, sure, but the same files that were there before, are there today underneath the new directory. We absolutely can know exactly what was written because in our use case, we just wrote it! It's working like that today. Run with current head and see if all of the files are removed. Run again with parallel and threads and check it again. It does work.